packaging/irods_prc_tests/test_rule_engine_plugin_metadata_guard_atomic.py

```diff
@@ -0,0 +1,126 @@
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.restore_irods_config()
+
+    def test_bad_config(self):
+        # set metadata_guard config AVU on collection to nonsense
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            coll = local_admin.collections.get("/tempZone")
+            coll.metadata['irods::metadata_guard'] = iRODSMeta('irods::metadata_guard', '{"broken": ["irods::"], ')
+
+        with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+            obj = unprivileged_user.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::atr1','val1')))
+
+            # bad config means operation should succeed
+            self.assertEqual(obj.metadata['irods::atr1'], iRODSMeta('irods::atr1', 'val1'))
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='remove', avu=iRODSMeta('irods::atr1','val1')))
+
+    def test_guard_atomic_operations_admin_only(self):
+        # set metadata_guard config AVU on collection to admin_only and irods:: as protected prefix
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            coll = local_admin.collections.get("/tempZone")
+            METADATA_GUARD_CONFIG = {
+                'admin_only': True,
+                'prefixes': ['irods::']
+            }
+            coll.metadata['irods::metadata_guard'] = iRODSMeta('irods::metadata_guard', json.dumps(METADATA_GUARD_CONFIG))
+
+        with iRODSSession(host='localhost', port=1247, user='alice', password='test', zone='tempZone') as unprivileged_user:
+            obj = unprivileged_user.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+
+            # "unguarded::" not protected, so operation should succeed
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('unguarded::atr1','val1')))
+            self.assertEqual(obj.metadata['unguarded::atr1'], iRODSMeta('unguarded::atr1', 'val1'))
+
+            # "irods::" protected, so operation should fail
+            self.assertRaises(irods.exception.CAT_INSUFFICIENT_PRIVILEGE_LEVEL, lambda: obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::atr','val'))))
+
+        with iRODSSession(host='localhost', port=1247, user='rods', password='rods', zone='tempZone') as local_admin:
+            obj = local_admin.data_objects.get(self.TEST_DATA_OBJECT_PATH)
+
+            # admin should be allowed to modify "irods::"
+            obj.metadata.apply_atomic_operations( AVUOperation(operation='add', avu=iRODSMeta('irods::atr1','val1')))
+            self.assertEqual(obj.metadata['irods::atr1'], iRODSMeta('irods::atr1', 'val1'))
+
+    def test_guard_atomic_operations_editor_list(self):
```

Comment:
Append issue number to test name.

The convention for that is `<test_name>__issue_12345`.